### PR TITLE
0.29 changelog updates

### PR DESCRIPTION
**Pull request by trbromley**
These are some quick final changes before the release.

### CODE REVIEW

File: doc/releases/changelog-0.29.0.md [hunk 1]
```diff
@@ -111,7 +111,10 @@
   ```
 
   This operation can be differentiated with hardware-compatible methods like parameter shifts
-  and it supports parameter broadcasting/batching, but not both at the same time.
+  and it supports parameter broadcasting/batching, but not both at the same time. Learn more by
```
Comment (rmoyard) [inline doc/releases/changelog-0.29.0.md:114]: Is it no the same, just we have two different implementations?

File: doc/releases/changelog-0.29.0.md [hunk 2]
```diff
@@ -313,6 +312,8 @@
            [0.92, 1.  , 0.91, 1.  ]]], requires_grad=True)
   ```
 
+  Note that we have extracted the first probability vector entry for each 100 shot evaluation.
```
Comment (rmoyard) [inline doc/releases/changelog-0.29.0.md:315]:
```suggestion
  Note that we have extracted the first probability vector entry for each 100 shot evaluations.
```

File: doc/releases/changelog-0.29.0.md [hunk 3]
```diff
@@ -501,7 +503,7 @@
 
 <h4>Operations and batching</h4>
 
-* A new function called `qml.dot` has been added to compute the dot product between a vector and a list of operators.
+* The function `qml.dot` has been updated to compute the dot product between a vector and a list of operators.
```
Comment (rmoyard) [inline doc/releases/changelog-0.29.0.md:506]: I am not sure we had it before?

File: doc/releases/changelog-0.29.0.md [hunk 4]
```diff
@@ -937,7 +940,7 @@
 * Only converts input to `qml.Hermitian` to a numpy array if the input is a list.
   [(#3820)](https://github.com/PennyLaneAI/pennylane/pull/3820)
 
-<h3>Contributors</h3>
+<h3>Contributors ✍</h3>
```
Comment (rmoyard) [inline doc/releases/changelog-0.29.0.md:943]: I think here you can delete the returns and add a period to the end.

**Review by rmoyard — COMMENTED** [review]
Thanks great job, a couple of comments :+1:

File: doc/releases/changelog-0.29.0.md (hunk 3 above)
Comment (trbromley) [reply doc/releases/changelog-0.29.0.md:506]: Yes it was there but for collections: https://docs.pennylane.ai/en/stable/code/api/pennylane.dot.html

File: doc/releases/changelog-0.29.0.md (hunk 1 above)
Comment (trbromley) [reply doc/releases/changelog-0.29.0.md:114]: I'll have to defer to @dwierichs for an answer!

File: doc/releases/changelog-0.29.0.md (hunk 2 above)
Comment (trbromley) [reply doc/releases/changelog-0.29.0.md:315]: :thinking: For me the non plural reads correctly.

Comment (timmysilv) [inline doc/releases/changelog-0.29.0.md:315]: I think a hyphen might work here then?
```suggestion
  Note that we have extracted the first probability vector entry for each 100-shot evaluation.
```

Comment (albi3ro) [reply doc/releases/changelog-0.29.0.md:315]: "for each evaluation".  Think the singular is right.

File: doc/releases/changelog-0.29.0.md (hunk 1 above)
Comment (dwierichs) [reply doc/releases/changelog-0.29.0.md:114]: I'm not sure I understand the question right, but the support table is
|                         | scalar parameters | broadcasted parameters |
|-------------------------|-------------------|------------------------|
| auto-differentiable     |         yes          |         yes               |
| hardware-differentiable |         yes          |             no           |

File: doc/releases/changelog-0.29.0.md [hunk 5]
```diff
@@ -281,8 +283,7 @@
   In addition they support batching, for example for quantum kernel execution with shot vectors.
```
Comment (timmysilv) [inline doc/releases/changelog-0.29.0.md:283]:
```suggestion
  In addition, they support batching, for example for quantum kernel execution with shot vectors.
```

**Review by timmysilv — COMMENTED** [review]
- line 254 says "the `qchem` workflow". does that make sense to users? I would have expected "module" instead of "workflow" here personally.

Some entries had duplicates, I've removed the second of each in this PR
- lines 553 and 614 have the same entry about `Sum` and `Prod` broadcasted params
- same for qnode kwarg validation, lines 636 and 673
- same for some experimental device API entries

File: doc/releases/changelog-0.29.0.md (hunk 4 above)
Comment (trbromley) [reply doc/releases/changelog-0.29.0.md:925]: It actually still renders nicely if we keep the names multiline:
image

**Review by timmysilv — APPROVED** [review]

**Review by rmoyard — APPROVED** [review]
Thanks :100:

**Pull request merged**